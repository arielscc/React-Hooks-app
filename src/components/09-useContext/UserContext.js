import { createContext } from 'react';

export const UserContext = createContext();

UserContext.displayName = 'ContextoDeUsers';

// export default UserContext;
