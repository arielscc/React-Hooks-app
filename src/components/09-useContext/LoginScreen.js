import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const newProps = {
    id: 10,
    user: 'Ariel',
  };

  return (
    <div>
      <h1>LoginScreen</h1>

      <button onClick={() => setUser(newProps)}>cambio</button>
    </div>
  );
};
