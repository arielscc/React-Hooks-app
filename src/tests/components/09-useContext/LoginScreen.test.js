import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en componente LoginScreen', () => {
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de llamarse el setUser con el argumento de newProps ', () => {
    wrapper.find('button').simulate('click');

    expect(setUser).toHaveBeenCalledWith({
      id: 10,
      user: 'Ariel',
    });
  });
});
