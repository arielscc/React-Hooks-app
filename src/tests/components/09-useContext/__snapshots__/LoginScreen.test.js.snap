// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente LoginScreen debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <button
    onClick={[Function]}
  >
    cambio
  </button>
</div>
`;
